libs/math/include/math/tensor.hpp

```diff
@@ -1295,10 +1305,12 @@ template <typename T, typename C>
 template <typename... Args>
 void Tensor<T, C>::Set(Args... args)
 {
-  assert(sizeof...(args) == stride_.size() + 1);  // Plus one as last arg is value
+  // Plus one as last arg is value
   if (sizeof...(args) != (stride_.size() + 1))
   {
-    throw exceptions::WrongIndices("too many or not enough indices given to Tensor::Set");
+    throw exceptions::WrongIndices(
+        "Wrong arguments quantity (" + std::to_string(stride_.size() + 1) +
+        ") given to Tensor::Set, expected: " + std::to_string(stride_.size() + 1));
```

Comment:
the string should print the expected and actual size - it now prints actual twice.

Reply:
done